pkg/cmd/openshift-controller-manager/openshiftcontrolplane_config.go

```diff
@@ -3,26 +3,35 @@ package openshift_controller_manager
 import (
 	"time"
 
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
 	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
 	"github.com/openshift/library-go/pkg/config/configdefaults"
-	"github.com/openshift/library-go/pkg/config/helpers"
-	leaderelectionconverter "github.com/openshift/library-go/pkg/config/leaderelection"
 )
 
+func asOpenshiftControllerManagerConfig(config *unstructured.Unstructured) (*openshiftcontrolplanev1.OpenShiftControllerManagerConfig, error) {
+	// Decode
+	result := &openshiftcontrolplanev1.OpenShiftControllerManagerConfig{}
+	if config != nil {
+		// make a copy we can mutate
+		configCopy := config.DeepCopy()
+		// force the config to our version to read it
+		configCopy.SetGroupVersionKind(openshiftcontrolplanev1.GroupVersion.WithKind("OpenShiftControllerManagerConfig"))
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(configCopy.Object, result); err != nil {
+			return nil, err
+		}
+	}
+
+	setRecommendedOpenShiftControllerConfigDefaults(result)
+	return result, nil
+}
+
 func setRecommendedOpenShiftControllerConfigDefaults(config *openshiftcontrolplanev1.OpenShiftControllerManagerConfig) {
-	configdefaults.SetRecommendedHTTPServingInfoDefaults(config.ServingInfo)
 	configdefaults.SetRecommendedKubeClientConfigDefaults(&config.KubeClientConfig)
-	config.LeaderElection = leaderelectionconverter.LeaderElectionDefaulting(config.LeaderElection, "kube-system", "openshift-master-controllers")
 
 	configdefaults.DefaultStringSlice(&config.Controllers, []string{"*"})
 
-	if config.ImageImport.MaxScheduledImageImportsPerMinute == 0 {
-		config.ImageImport.MaxScheduledImageImportsPerMinute = 60
-	}
-	if config.ImageImport.ScheduledImageImportMinimumIntervalSeconds == 0 {
-		config.ImageImport.ScheduledImageImportMinimumIntervalSeconds = 15 * 60
-	}
-
 	configdefaults.DefaultString(&config.SecurityAllocator.UIDAllocatorRange, "1000000000-1999999999/10000")
```

Comment:
Technically some of the other defaulting could be removed, as it these are remains of other controllers being included in OCM. But let's not do it now as it is not connected to the library-go refactoring.